src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -693,20 +691,37 @@ def show_default_diagnostics_configuration(is_windows_os=False):
     '''show the default config file which defines data to be collected'''
     return get_default_diag_config(is_windows_os)
 
-def vm_add_nics(resource_group_name, vm_name, nic_ids=None, nic_names=None, primary_nic=None):
-    '''add network interface configurations to the virtual machine
+def vm_show_nic(resource_group_name, vm_name, nic=None):
+    ''' Show details of a network interface configuration attached to a virtual machine '''
+    vm = _vm_get(resource_group_name, vm_name)
+    found = next((n for n in vm.network_profile.network_interfaces if nic in n.id), None) # pylint: disable=no-member
```

Comment:
`nic in n.id` should consider case insensitive matching. Also is `nic` always a id? If not, then `in` is not a reliable matching as it is possible the nic name will collide with any sub-str in the id.